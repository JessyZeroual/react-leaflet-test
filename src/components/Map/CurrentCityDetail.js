import React from "react";
import {
  StyledWrapperCurrentCityDetail,
  StyledCurrentCityDetailText,
} from "./Map.styled";

const CurrentCityDetail = ({ currentCity }) => (
  <>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/buildings.svg" width="30" alt="icon city" />
      <StyledCurrentCityDetailText>
        {currentCity.city}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/flag.svg" width="30" alt="icon state" />
      <StyledCurrentCityDetailText>
        {currentCity.state}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/growth.svg" width="30" alt="icon growth" />

      <StyledCurrentCityDetailText
        isNegative={Math.sign(currentCity.growth_from_2000_to_2013) === -1}
      >
        {`${currentCity.growth_from_2000_to_2013}%`}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/population.svg" width="30" alt="icon population" />
      <StyledCurrentCityDetailText>
        {currentCity.population.toLocaleString("en-US")}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
  </>
);

export default CurrentCityDetail;
