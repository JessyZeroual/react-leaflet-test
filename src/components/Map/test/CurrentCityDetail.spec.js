import React from "react";
import TestRenderer from "react-test-renderer";
import CurrentCityDetail from "../CurrentCityDetail";

describe("CurrentCityDetail", () => {
  describe("When CurrentCityDetail is mounted", () => {
    const renderer = TestRenderer.create(
      <CurrentCityDetail
        currentCity={{
          city: "New York",
          growth_from_2000_to_2013: 4.8,
          latitude: 40.7127837,
          longitude: -74.0059413,
          population: 8405837,
          rank: 1,
          state: "New York",
        }}
      />
    );

    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("receives correct properties", () => {
      const { currentCity } = renderer.root.findByType(CurrentCityDetail).props;

      expect(typeof currentCity.city).toEqual("string");
      expect(typeof currentCity.growth_from_2000_to_2013).toEqual("number");
      expect(typeof currentCity.latitude).toEqual("number");
      expect(typeof currentCity.longitude).toEqual("number");
      expect(typeof currentCity.population).toEqual("number");
      expect(typeof currentCity.rank).toEqual("number");
      expect(typeof currentCity.state).toEqual("string");
    });
  });
});
