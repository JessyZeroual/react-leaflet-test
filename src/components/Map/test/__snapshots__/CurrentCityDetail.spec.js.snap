// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCityDetail When CurrentCityDetail is mounted renders correctly 1`] = `
Array [
  <div
    className="sc-AxirZ bnSjNE"
  >
    <img
      alt="icon city"
      src="/icon/buildings.svg"
      width="30"
    />
    <p
      className="sc-AxiKw gnjXWm"
    >
      New York
    </p>
  </div>,
  <div
    className="sc-AxirZ bnSjNE"
  >
    <img
      alt="icon state"
      src="/icon/flag.svg"
      width="30"
    />
    <p
      className="sc-AxiKw gnjXWm"
    >
      New York
    </p>
  </div>,
  <div
    className="sc-AxirZ bnSjNE"
  >
    <img
      alt="icon growth"
      src="/icon/growth.svg"
      width="30"
    />
    <p
      className="sc-AxiKw gnjXWm"
    >
      4.8%
    </p>
  </div>,
  <div
    className="sc-AxirZ bnSjNE"
  >
    <img
      alt="icon population"
      src="/icon/population.svg"
      width="30"
    />
    <p
      className="sc-AxiKw gnjXWm"
    >
      8,405,837
    </p>
  </div>,
]
`;
